src/dos/dos_keyboard_layout.cpp

```diff
@@ -1639,8 +1639,13 @@ KeyboardErrorCode DOS_LoadKeyboardLayoutFromLanguage(const char * language_pref)
 		            language.c_str());
 	}
 	// Regardless of the above, carry on with setting up the layout
+	std::string layout = language;
+	static constexpr std::array<std::string_view, 1> exceptions{"nl"};
+	if (std::find(std::begin(exceptions), std::end(exceptions), layout) !=
+	    std::end(exceptions))
+		layout = "";
 	const auto codepage = lookup_codepage_from_country(country);
-	const auto result   = DOS_LoadKeyboardLayout(language.c_str(), codepage, "auto");
+	const auto result = DOS_LoadKeyboardLayout(layout.c_str(), codepage, "auto");
```

Comment:
The the body of this `DOS_LoadKeyboardLayoutFromLanguage()` function makes use of lookup routines, like:
 - `lookup_country_from_code`
 - `lookup_codepage_from_country`

I think having another lookup routine that handles these language layout exceptions (or mappings), would be a nice place to manage these (and likely more to come).

It could be something like: `maybe_get_language_to_layout_exception(language)`, and it would return a `const char *`, that we could use straight-away in the `DOS_LoadKeyboardLayout` like you've done here (nicely named, `layout`).

I suppose the `maybe_get_language_to_layout_exception` function could hold a static `std::map` of language and layout strings.

For each mapping, we could include a comment justifying the entry, like what @rderooy explained in this case.

Reply:
Thanks for the suggestion. There is now a map for such exceptions, along with the `lookup_language_to_layout_exception` function as suggested above.